rcongui/src/components/Scoreboard/GamesScore.js

```diff
@@ -138,6 +165,13 @@ const GamesScore = ({ classes }) => {
     if (!currentMapId) {
       return;
     }
+    if (!parseInt(slug)) {
+      window.location.hash = `#/gamescoreboard/${currentMapId}`
+      console.log(`No ID supplied, redirecting to ${window.location.hash}`);
+    } else {
+      setCurrentMapId(parseInt(slug));
```

Comment:
You should add the slug in the list of dependancies on line 182, this will cause this function to re-excute if the slug changes. And therefore you won't need the doSelectMap at all. We just need links.